build.gradle.kts

```diff
@@ -30,23 +28,80 @@ dependencies {
+                    implementation("org.springframework.boot:spring-boot-starter-validation")
+                    implementation("com.fasterxml.jackson.module:jackson-module-kotlin")
 
-  testImplementation("org.springframework.boot:spring-boot-starter-test")
-  testImplementation("com.ekino.oss.jcv:jcv-hamcrest:$jcvVersion")
-  testImplementation("com.willowtreeapps.assertk:assertk-jvm:$assertkVersion")
+                    implementation("org.springframework.boot:spring-boot-starter-test")
+                    implementation("com.ekino.oss.jcv:jcv-hamcrest:1.5.0")
+                    implementation("com.willowtreeapps.assertk:assertk-jvm:0.25")
+                }
+            }
+        }
 
-  securityTestImplementation("org.springframework.security:spring-security-web")
+        val securityTest by registering(JvmTestSuite::class) {
+            dependencies {
+                implementation(project())
+                implementation("org.springframework.security:spring-security-web")
+            }
+            sources {
+                java {
+                    setSrcDirs(listOf("src/securityTest/kotlin"))
+                }
+            }
+        }
 
-  dataRestTestImplementation("org.springframework.boot:spring-boot-starter-data-rest")
+        val dataRestTest by registering(JvmTestSuite::class) {
+            dependencies {
+                implementation(project())
+                implementation("org.springframework.boot:spring-boot-starter-data-rest")
+            }
+            sources {
+                java {
+                    setSrcDirs(listOf("src/dataRestTest/kotlin"))
+                }
+            }
+        }
 
-  txTestImplementation("org.springframework.boot:spring-boot-starter-data-jpa")
+        val txTest by registering(JvmTestSuite::class) {
+            dependencies {
+                implementation(project())
+                implementation("org.springframework.boot:spring-boot-starter-data-jpa")
+            }
+            sources {
+                java {
+                    setSrcDirs(listOf("src/txTest/kotlin"))
+                }
+            }
+        }
 
-  awsTestImplementation("software.amazon.awssdk:s3:$awsSdkVersion")
+        val awsTest by registering(JvmTestSuite::class) {
+            dependencies {
+                implementation(project())
+                implementation("software.amazon.awssdk:s3:$awsSdkVersion")
+            }
+            sources {
+                java {
+                    setSrcDirs(listOf("src/awsTest/kotlin"))
+                }
+            }
+        }
+    }
 }
 
-configurations {
-  all {
-    exclude(module = "mockito-core")
-    exclude(module = "assertj-core")
-  }
+tasks.named("build") {
+    dependsOn(testing.suites.named("securityTest"))
+    dependsOn(testing.suites.named("dataRestTest"))
+    dependsOn(testing.suites.named("txTest"))
+    dependsOn(testing.suites.named("awsTest"))
```

Comment:
what would you think of `testing.suites.forEach(::dependsOn)` instead ?

Reply:
Nice catch, done.